src/index.ts

```diff
@@ -67,42 +68,59 @@ extendEnvironment((hre) => {
   })
 })
 
-export const TASK_PREPARE_ARTIFACTS = "prepare-artifacts"
+extendConfig(
+  (config: HardhatConfig, userConfig: Readonly<HardhatUserConfig>) => {
+    const exportUserConfig = userConfig.deploymentArtifactsExport
 
-task(TASK_PREPARE_ARTIFACTS)
-  .addOptionalPositionalParam(
-    "destination",
-    "destination folder where the artifacts files will be written to",
-    "artifacts",
-    types.string
-  )
-  .setAction(async (args, hre) => {
-    const sourceDir = path.join(hre.config.paths.deployments, hre.network.name)
-    const destinationDir = path.join(hre.config.paths.root, args.destination)
-
-    console.log(
-      `Preparing deployment artifacts for network ${hre.network.name}...`
-    )
-
-    if (!fs.pathExistsSync(sourceDir)) {
-      throw new HardhatPluginError(
-        "@keep-network/hardhat-helpers",
-        `source deployments artifacts directory doesn't exist [${sourceDir}]`
+    if (config.deploymentArtifactsExport === undefined) {
+      config.deploymentArtifactsExport = {}
+    }
+
+    let defaultDestinationDir = exportUserConfig?.default ?? "artifacts"
+
+    if (path.isAbsolute(defaultDestinationDir)) {
+      defaultDestinationDir = defaultDestinationDir
+    } else {
+      defaultDestinationDir = path.normalize(
+        path.join(config.paths.root, defaultDestinationDir)
       )
     }
 
-    console.debug(
-      `  Source:      ${sourceDir}\n  Destination: ${destinationDir}`
-    )
+    const networks = Object.keys(config.networks)
 
-    fs.ensureDirSync(destinationDir)
-    fs.emptyDirSync(destinationDir)
+    networks.forEach((value: string) => {
+      if (
+        exportUserConfig === undefined ||
+        exportUserConfig[value] === undefined
+      ) {
+        config.deploymentArtifactsExport[value] = defaultDestinationDir
+      } else {
+        let networkDestinationDir = exportUserConfig[value]!
 
-    fs.copySync(sourceDir, destinationDir, {
-      recursive: true,
+        if (path.isAbsolute(networkDestinationDir)) {
+          networkDestinationDir = networkDestinationDir
+        } else {
+          networkDestinationDir = path.normalize(
+            path.join(config.paths.root, networkDestinationDir)
+          )
+        }
+      }
     })
+  }
+)
 
-    // TODO: Remove address for `hardhat` network.
+export const TASK_EXPORT_DEPLOYMENT_ARTIFACTS = "export-deployment-artifacts"
 
-    console.log(`${emoji("🙌 ")}Done!`)
-  })
+task(TASK_EXPORT_DEPLOYMENT_ARTIFACTS).setAction(async (args, hre) => {
```

Comment:
![Screenshot 2023-04-04 at 15 28 29](https://user-images.githubusercontent.com/3156420/229808452-703a7303-abee-4329-ad9b-d10ad74892c3.png)

Can we add a description?

Reply:
cffe2cc992321990bc0598fd4d85d5bd61751ca0